### PR TITLE
small changes so that it can be used directly in a npm pipeline

### PR DESCRIPTION
**Pull request by alex-milanov**
usage in a node module
```js
const prettify = require('code-prettify')
// default
prettify.prettyPrint();
// single use
document.querySelector('pre > code').innerHTML = prettify.prettyPrintOne('function() {console.log('foo');}');
```
usage in a task automation (like gulp, grunt...) for css preprocessor (like sass, less)
```js
require('code-prettify').includePaths
```
then inside a sass file
```sass
@import 'prettify'
```

### CODE REVIEW

**Comment by alex-milanov** [comment]
I signed it!

**Comment by alex-milanov** [comment]
Just some updates.
 
I've successfully used my fork as a npm package (`npm i -S alex-milanov/code-prettify`) with browserify for a simple code editor for the slides app for a local talk I did on functional **reactive javascript**.

You can check:
- the repo at https://github.com/alex-milanov/first-steps-frp-js
- and demo https://alex-milanov.github.io/first-steps-frp-js

Examples of usage:
- code editor with console https://github.com/alex-milanov/first-steps-frp-js/blob/master/src/js/ui/slides/code/index.js
- the paths used in the asset pipeline https://github.com/alex-milanov/first-steps-frp-js/blob/master/bin/sass-paths.js
- in the main sass file https://github.com/alex-milanov/first-steps-frp-js/blob/master/src/sass/style.sass
- some style modification https://github.com/alex-milanov/first-steps-frp-js/blob/master/src/sass/ext/_prettify.sass

Can someone check if there are potential side effects from the changes I made so that we can merge the PR? If there are we can discuss them so that I can make the needed modifications. 

//cc @amroamroamro @mikesamuel

**Comment by alex-milanov** [comment]
btw. I created a gitter chat room for my fork at https://gitter.im/alex-milanov/code-prettify (You guys could create an "official" one or join the one I created, both options work for me)

**Comment by ateev** [comment]
@alex-milanov Thanks for this !  Why isn't this merged with the original project yet ?

**Comment by alex-milanov** [comment]
@ateev beats me.

//cc @amroamroamro @mikesamuel

**Comment by alex-milanov** [comment]
@yuschick seems you have to sign a google CLA (Contributor License Agreement).
Don't how relevant this is since this PR has been opened since Feb 18.

File: src/prettify.js [hunk 1]
```diff
@@ -1740,3 +1741,14 @@ var prettyPrint;
     });
   }
 })();
+
+// npm require support
+if (module && module.exports) module.exports = {
```
Comment (mikesamuel) [inline src/prettify.js]: Won't if(module) throw an exception if there is no global named "module".

Maybe if (typeof module !== 'undefined' && module.exports)

File: src/prettify.js [hunk 2]
```diff
@@ -141,7 +141,8 @@ var PR;
  * UI events.
  * If set to {@code false}, {@code prettyPrint()} is synchronous.
  */
-window['PR_SHOULD_USE_CONTINUATION'] = true;
+var PR_SHOULD_USE_CONTINUATION = true;
+if (typeof window !== 'undefined') window.PR_SHOULD_USE_CONTINUATION = PR_SHOULD_USE_CONTINUATION;
```
Comment (alex-milanov) [inline src/prettify.js]: Sure. I will check it out. btw. wouldn't it be more logical to have the sources in the `src` folder and the destination in a `dist` one?

File: src/prettify.js (hunk 1 above)
Comment (alex-milanov) [inline src/prettify.js]: Probably it shouldn't be an issue in most cases but the second one would be more consistent.

File: src/prettify.js (hunk 2 above)
Comment (alex-milanov) [inline src/prettify.js]: @mikesamuel also aren't these vars constants -> can we use
```js
const PR_SHOULD_USE_CONTINUATION = true;
```

**Comment by alex-milanov** [comment]
@mikesamuel after struggling for a while with the closure compile process and stumbling upon how run_prettify is formed found a way to add node support by introducing `js-modues/node_prettify.js` and modifying the `preprocess` and `copy:prettify` grunt tasks. 
Also pointed the main in package.json (used only in node I hope) to `src/node_prettify.js`
//cc @yuschick Can you confirm it also works in your setup?

**Comment by alex-milanov** [comment]
@mikesamuel I hope we won't have to wait 6 more months for a response :-)

**Comment by evolve2k** [comment]
Bump.

**Comment by evolve2k** [comment]
Looks good to me. Let's merge this in.

**Comment by mikesamuel** [comment]
Thanks for doing this.  Sorry for taking so long to get back to it.

**Comment by evolve2k** [comment]
Thank you everyone!!! 🙌🎉

**Comment by evolve2k** [comment]
Any possibility of doing a new public release to help get this feature out into the wild? 🦁

**Comment by alex-milanov** [comment]
If you need help I am available

**Comment by Pyronerd62** [comment]
How does this work. 
I installed the npm via this 
```
npm i git://github.com/google/code-prettify#master -- save
```

```
const prettify = require('code-prettify')
// default
prettify.prettyPrint();
```

This is the error it responds with.
`TypeError: prettify.prettyPrint is not a function`

What do you have to do to use this as a module?

**Pull request merged**